util/updateXML/sources/Model/Data/Model.py

```diff
@@ -0,0 +1,194 @@
+from abc import ABC
+import lxml.etree
+
+from ...Common import *
+from ...UpdateXMLExceptions import *
+
+
+class Model(ABC):
+    """
+    This abstract class defines a common interface for classes which own a parset to :
+        - add parameters/references in parset
+        - remove parameters/references from parset
+        - change parameters/references name
+
+    Attribute
+    ----------
+    parset_ : str
+        parset content
+    """
+    def __init__(self, parset):
+        self.parset_ = parset
+
+    # ---------------------------------------------------------------
+    #   USER METHODS
+    # ---------------------------------------------------------------
+
+    def add_param(self, param_type, param_name, param_value):
+        """
+        Add a parameter in a specific parset in a Par file
+
+        Parameters:
+            param_type (str): type of the parameter to add
+            param_name (str): name of the parameter to add
+            param_value (str): value of the parameter to add
+        """
+        params_with_same_name = self.parset_.xpath('./dyn:par[@name="' + param_name + '"]', namespaces=NAMESPACE_URI)
+        if len(params_with_same_name) == 0:
+            if len(self.parset_) == 0:
+                self.parset_.text = XML_DOUBLE_INDENTATION
+            else:
+                self.parset_[-1].tail += DOUBLE_SPACE
+            par_elem = lxml.etree.SubElement(self.parset_,
+                                                xmlns(XML_PARAMETER),
+                                                {'type': param_type, 'name': param_name, 'value': param_value})
+            par_elem.tail = XML_SIMPLE_INDENTATION
+        elif len(params_with_same_name) >= 2:
+            raise DuplicateParameters(param_name, self.parset_.attrib['id'], self.parset_.base)
```

Comment:
WHy do we have an error here? if the parameter aready exists we should leave it this way

Reply:
A parameter/reference is supposed to only appear once in a parset, not twice or more. This exception is triggered if there is this error.